### PR TITLE
Improvement of FileUtils test coverage

### PR DESCRIPTION
**Pull request by Kled9**

### CODE REVIEW

**Comment by garydgregory** [comment]
-1: @Kled9 The way to test assertions in JUnit 5 is with `assertThrows`

**Comment by Kled9** [comment]
@garydgregory Ok thanks for the information, I modified the tests, now they use `assertThrows` from JUnit5.

File: src/test/java/org/apache/commons/io/FileUtilsTestCase.java [hunk 1]
```diff
@@ -244,6 +244,19 @@ public void test_openInputStream_notExists() throws Exception {
         }
     }
 
+    @Test
+    public void openInputStreamCannotReadFile() throws IOException {
+        final File file = new File(temporaryFolder, "test.txt") {
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/FileUtilsTestCase.java:249]: Use Mockito for this type of test fixture.

File: src/test/java/org/apache/commons/io/FileUtilsTestCase.java [hunk 2]
```diff
@@ -321,6 +334,19 @@ public void test_openOutputStream_notExistsCannotCreate() throws Exception {
         }
     }
 
+    @Test
+    public void openOutputStreamCannotWriteFile() throws Exception {
+        final File file = new File(temporaryFolder, "test.txt") {
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/FileUtilsTestCase.java:339]: Use Mockito for this type of test fixture.

File: src/test/java/org/apache/commons/io/FileUtilsTestCase.java [hunk 3]
```diff
@@ -1237,6 +1263,50 @@ public void testCopyFile2WithoutFileDatePreservation() throws Exception {
...
+        assertThrows(IOException.class, () -> FileUtils.copyFile(source, destination, false));
+    }
+
+    @Test
+    public void copyFileFailedWhenDestinationParentDirectoryCannotBeCreated() {
+        final File destination = new File(temporaryFolder, "destination/drop.txt") {
+            @Override
+            public File getParentFile() {
+                final File parentFile = new File(String.valueOf(super.getParentFile())) {
+                    @Override
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/FileUtilsTestCase.java:1283]: Use Mockito for this type of test fixture.

File: src/test/java/org/apache/commons/io/FileUtilsTestCase.java [hunk 4]
```diff
@@ -1237,6 +1263,50 @@ public void testCopyFile2WithoutFileDatePreservation() throws Exception {
...
+            }
+        };
+
+        assertThrows(IOException.class, () -> FileUtils.copyFile(testFile1, destination, false));
+    }
+
+    @Test
+    public void copyFileFailedWhenDestinationExistsButIsReadyOnly() {
+        final File destination = new File(temporaryFolder, "destination/drop.txt") {
+            @Override
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/FileUtilsTestCase.java:1298]: Use Mockito for this type of test fixture.

File: src/test/java/org/apache/commons/io/FileUtilsTestCase.java [hunk 5]
```diff
@@ -2507,6 +2597,21 @@ public void testDeleteDirectoryWithNonDirectory() throws Exception {
         }
     }
 
+    @Test
+    public void deleteDirectoryFailed() {
+        File directoryToDelete = new File(temporaryFolder, "directoryToDelete") {
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/io/FileUtilsTestCase.java:2602]: Use final where posssible,
Use Mockito for this type of test fixture.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @Kled9 
Please see my comments.

**Comment by garydgregory** [comment]
Closing, no update in over a year.

**Pull request closed without merging**